include/modem/sms.h

```diff
@@ -24,13 +24,91 @@ extern "C" {
 #include <zephyr/types.h>
 #include <sys/types.h>
 
+enum sms_type {
+	SMS_TYPE_DELIVER = 0,
+	SMS_TYPE_SUBMIT_REPORT
+};
+
 /** @brief SMS PDU data. */
 struct sms_data {
+	enum sms_type type;
 	char *alpha;
 	uint16_t length;
 	char *pdu;
 };
 
+enum sms_deliver_alphabet {
+	GSM_ENCODING_8BIT,
+	GSM_ENCODING_UCS2,
+};
+
+enum sms_deliver_class {
+	GSM_CLASS0,
+	GSM_CLASS1,
+	GSM_CLASS2,
+	GSM_CLASS3,
+};
+
+struct sms_deliver_time {
+	uint8_t year;
+	uint8_t month;
+	uint8_t day;
+	uint8_t hour;
+	uint8_t minute;
+	uint8_t second;
+	int8_t timezone;
+};
+
+struct sms_deliver_address {
+	uint8_t address[12];
+	uint8_t length;
+	uint8_t type;
+};
+
+struct sms_udh_concatenated {
+	bool present;
+	uint16_t ref_number;
+	uint8_t total_msgs;
+	uint8_t seq_number;
+};
+
+struct sms_udh_app_port {
+	bool present;
+	uint16_t dest_port;
+	uint16_t src_port;
+};
+
+struct sms_deliver_header {
```

Comment:
We need to see whether this content for the header is fine, whether to remove some fields not likely needed by the client, or whether to have all header information included as parsed into structures defined in sms_deliver.c

Reply:
This has been modified based on discussions